utils.py

```diff
@@ -83,44 +88,102 @@ def MakePlots(truth_XY, predict_XY, PUPPI_XY, path_out):
     plt.ylabel('<MET Estimation>/<MET Truth>')
     plt.savefig(f"{path_out}MET_response.png")
     plt.close()
-    # response correction factors
-    sfs_truth    = np.take(truth_means/truth_means,    np.digitize(truth_PtPhi[:,0], binnings)-1, mode='clip')
-    sfs_predict  = np.take(predict_means/truth_means,  np.digitize(truth_PtPhi[:,0], binnings)-1, mode='clip')
-    sfs_PUPPI = np.take(PUPPI_means/truth_means, np.digitize(truth_PtPhi[:,0], binnings)-1, mode='clip')
-    # resolution defined as (q84-q16)/2.0
+    
     def resolqt(y):
         return(np.percentile(y,84)-np.percentile(y,16))/2.0
-    bin_resolX_predict, bin_edges, binnumber = binned_statistic(truth_PtPhi[:,0], truth_XY[:,0] - predict_XY[:,0] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
-    bin_resolY_predict, _, _                 = binned_statistic(truth_PtPhi[:,0], truth_XY[:,1] - predict_XY[:,1] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
-    bin_resolX_PUPPI, _, _                = binned_statistic(truth_PtPhi[:,0], truth_XY[:,0] - PUPPI_XY[:,0] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
-    bin_resolY_PUPPI, _, _                = binned_statistic(truth_PtPhi[:,0], truth_XY[:,1] - PUPPI_XY[:,1] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
+
+    def resolqt_upper(y):
+        return(np.percentile(y,90)-np.percentile(y,10))/2.0
+
+    def resolqt_lower(y):
+        return(np.percentile(y,76)-np.percentile(y,22))/2.0
```

Comment:
what are upper and lower trying to compute here?

the original function is trying to get the "1 sigma" error band [16%, 84%] / 2 (i.e. +/-1sigma roughly corresponds to the middle 68%)

if you were trying to get the "2 sigma" error band, you could do [2.5%, 97.5%] / 2

but these other values seem somewhat random to me?

you could also combine all of these into 1 function with parameters

Reply:
Yeah, if the +-6 percentile points was random— I tried 2sigma  first, but the error bars were so large, it didn't seem that useful. Ill share a plot with you

Reply:
They calculate the width of the distribution at slightly narrow and wider points in the distribution. Is there a better way to calculate error? +-6 from 1 SD is arbitrary— I tried 2 𝜎  at first, but the error bars are pretty large(upper bound is essentially double the resolution). If we do this, then what should I make the lower error bar? Width at half of a sigma? I can share a plot on slack.
For the `resolqt` function, it seems like `binned_statistic` only accepts functions with one parameter as a statistic, so I could use `partial(resolqt, upper=#, lower=#)` but this doesn't seem better than 3 functions.